ci-operator/step-registry/opendatahub-io/ci/s390x-build/dashboard/opendatahub-io-ci-s390x-build-dashboard-commands.sh

```diff
@@ -0,0 +1,139 @@
+            IMAGE_TAG="${RELEASE_VERSION}-${YEAR_INDEX}-${PULL_BASE_SHA:0:7}"
+            IMAGE_FLOATING_TAG="${RELEASE_VERSION}-${YEAR_INDEX}"
+            ;;
+        periodic)
+            echo "Building weekly image tag for a $JOB_TYPE job"
+            IMAGE_TAG="${RELEASE_VERSION}-weekly"
+            ;;
+        *)
+            echo "ERROR Cannot publish an image from a $JOB_TYPE job"
+            exit 1
+            ;;
+    esac
+fi
+
+# Build destination image reference
+DESTINATION_REGISTRY_REPO="$REGISTRY_HOST/$REGISTRY_ORG/$IMAGE_REPO"
+DESTINATION_IMAGE_REF="$DESTINATION_REGISTRY_REPO:$IMAGE_TAG"
+if [[ -n "${IMAGE_FLOATING_TAG-}" ]]; then
+    FLOATING_IMAGE_REF="$DESTINATION_REGISTRY_REPO:$IMAGE_FLOATING_TAG"
+    DESTINATION_IMAGE_REF="$DESTINATION_IMAGE_REF $FLOATING_IMAGE_REF"
+fi
+
+echo "Image is $DESTINATION_IMAGE_REF"
+echo "commit hash is ${PULL_BASE_SHA:0:7}"
+echo "Entire commit has is $PULL_BASE_SHA"
+echo "JOB SPECS are $JOB_SPEC"
+
+# Initialize ALL_VARS as an array of variable assignments
+ALL_VARS=("DESTINATION_IMAGE_REF='$DESTINATION_IMAGE_REF' PULL_BASE_SHA='$PULL_BASE_SHA' SECRETS_PATH='$SECRETS_PATH' REGISTRY_SECRET_FILE='$REGISTRY_SECRET_FILE' REGISTRY_HOST='$REGISTRY_HOST' DOCKER_USER='$DOCKER_USER' DOCKER_PASS='$DOCKER_PASS' PLATFORMS='$PLATFORMS'")
+ALL_VARS_STR=$(IFS=" "; echo "${ALL_VARS[*]}")
+export ALL_VARS_STR
+
+# create ssh session to zvsi and pass the script
+ssh "${ssh_options[@]}" root@$zvsi_fip "$ALL_VARS_STR bash -s" << 'EOF'
+#Installing docker in zvsi
+echo "Installing docker engine in zvsi"
+
+# Add Docker's official GPG key:
+apt-get update
+apt-get install ca-certificates curl -y
+install -m 0755 -d /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources:
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+  $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
+  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+apt-get update
+apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin -y
+
+echo "Platforms is $PLATFORMS "
+echo "Image tag is $IMAGE_TAG "
+git clone https://github.com/opendatahub-io/odh-dashboard
+git checkout $PULL_BASE_SHA
+
+# Enable Buildx for multiarch builds
+docker buildx create --platform="${PLATFORMS}" --name mybuilder --use
+
+# Log in to the registry
+DOCKER_USER=$(cat "${SECRETS_PATH}/${REGISTRY_SECRET_FILE}" | jq -r ".auths[\"${REGISTRY_HOST}\"].auth" | base64 -d | cut -d':' -f1)
+DOCKER_PASS=$(cat "${SECRETS_PATH}/${REGISTRY_SECRET_FILE}" | jq -r ".auths[\"${REGISTRY_HOST}\"].auth" | base64 -d | cut -d':' -f2)
+
+docker login -u "${DOCKER_USER}" -p "${DOCKER_PASS}" "${REGISTRY_HOST}"
+
+# Build and push the multiarch image
+echo "pushing image $DESTINATION_IMAGE_REF "
```

Comment:
Ok so with this we achieve multiarch images with the rest of the arch plus s390x right?

Reply:
We reviewed this, apologies but it's been a couple of weeks and I don't recall this correctly.